crates/block-producer/src/block_producer.rs

```diff
@@ -385,17 +385,19 @@ impl BlockProducer {
 
         // get txs & withdrawal requests from mem pool
         let (opt_finalized_custodians, block_param) = {
+            let output_fut = {
+                let t = Instant::now();
+                log::debug!("[compose_next_block_submit_tx] acquire mem-pool",);
+                let mem_pool = self.mem_pool.lock().await;
+                log::debug!(
+                    "[compose_next_block_submit_tx] unlock mem-pool {}ms",
+                    t.elapsed().as_millis()
+                );
+                mem_pool.output_mem_block(&OutputParam::new(retry_count))
```

Comment:
Can I say that this pr is about wrapping the old logic of **output_mem_block** into a future?
Could you please elaborate on why this change reduces the lock time of the output block?

Reply:
`output_mem_block` produces a future. This PR moves mostly calculation job into this future, so we can release mem-pool early.

IMO we can move the packaging logic out of mem-pool, this can makes mem-pool logic more clear.